tests_e2e/tests/ext_update/ext_update_failure.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python3
+
+# Microsoft Azure Linux Agent
+#
+# Copyright 2018 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import uuid
+from typing import List, Dict, Any
+
+from assertpy import fail, assert_that
+
+from tests_e2e.tests.lib.agent_test import AgentVmTest
+from tests_e2e.tests.lib.logging import log
+from tests_e2e.tests.lib.ssh_client import SshClient
+from tests_e2e.tests.lib.virtual_machine_extension_client import VirtualMachineExtensionClient
+from tests_e2e.tests.lib.vm_extension_identifier import VmExtensionIds, VmExtensionIdentifier
+
+
+class ExtensionUpdateFailureTest(AgentVmTest):
+    """
+    This test verifies that the agent reports correct status and code when extension operation fails during the ext update.
+    """
+
+    def run(self):
+        ssh_client: SshClient = self._context.create_ssh_client()
+        is_arm64: bool = ssh_client.get_architecture() == "aarch64"
+
+        custom_script_2_0 = VirtualMachineExtensionClient(
+            self._context.vm,
+            VmExtensionIds.CustomScript,
+            resource_name="CustomScript")
+
+        if is_arm64:
+            log.info("Will skip the update scenario, since currently there is only 1 version of CSE on ARM64")
+            return
+        else:
+            log.info("Installing %s", custom_script_2_0)
+            message = f"Hello {uuid.uuid4()}!"
+            custom_script_2_0.enable(
+                protected_settings={
+                    'commandToExecute': f"echo \'{message}\'"
+                },
+                auto_upgrade_minor_version=False
+            )
+            custom_script_2_0.assert_instance_view(expected_version="2.0", expected_message=message)
+
+        log.info("Modifying existing handler commands (disable operation) in HandlerManifest.json to fail the disable operation during update")
+
+        output = ssh_client.run_command(f"ext_update-modify_handler_manifest.py --extension-name '{custom_script_2_0._identifier}' --cmd-name disableCommand --cmd-value 'disablefailed'", use_sudo=True)
```

Comment:
should we also update the manifest so continueOnUpdate is False?

Reply:
done